cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -319,9 +326,16 @@ var (
 				logger.WithField("paths", asPaths).Warn("Deselecting Application Server paths")
 				asPaths = nil
 			}
-			if len(jsPaths) > 0 && jsMismatch {
-				logger.WithField("paths", jsPaths).Warn("Deselecting Join Server paths")
-				jsPaths = nil
+
+			if len(jsPaths) > 0 {
+				if device.ClaimAuthenticationCode.GetValue() != "" {
+					// ClaimAuthenticationCode is already retrieved from the IS. We can unset the related JS paths.
+					jsPaths = ttnpb.ExcludeFields(jsPaths, claimAuthenticationCodePaths...)
+				}
+				if jsMismatch {
+					logger.WithField("paths", jsPaths).Warn("Deselecting Join Server paths")
+					jsPaths = nil
+				}
```

Comment:
This can still be empty because of the `ExcludeFlags` above. Why not move this after L318?